chpl/chpl-service/src/test/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/TestFunctionalityAllowedByRoleReviewerTest.java

```diff
@@ -0,0 +1,378 @@
+        Mockito.when(errorMessages.getMessage(ArgumentMatchers.anyString())).thenReturn(ERROR_MESSAGE);
+
+        permissions = Mockito.mock(ResourcePermissions.class);
+        Mockito.when(permissions.doesUserHaveRole(ArgumentMatchers.any(List.class))).thenReturn(true);
+
+        reviewer = new TestFunctionalityAllowedByRoleReviewer(ff4j, env, permissions, errorMessages);
+    }
+
+    @Test
+    public void review_BeforeEffectiveRuleDate_NoErrorMessages() {
+        // Setup
+        Mockito.when(ff4j.check(FeatureList.EFFECTIVE_RULE_DATE_PLUS_ONE_WEEK)).thenReturn(false);
+
+        CertifiedProductSearchDetails listing = CertifiedProductSearchDetails.builder().build();
+
+        // Run
+        reviewer.review(null, listing);
+
+        // Check
+        Assert.assertEquals(0, listing.getErrorMessages().size());;
+    }
+
+    @Test
+    public void review_AfterRuleDateAndTestFunctionalityDidNotChange_NoErrorMessages() {
+        // Setup
+        CertifiedProductSearchDetails existingListing = CertifiedProductSearchDetails.builder()
+                .certificationResult(CertificationResult.builder()
+                        .id(5L)
+                        .criterion(CertificationCriterion.builder()
+                                .id(17l)
+                                .certificationEdition("2015")
+                                .certificationEditionId(3L)
+                                .number("170.315 (b)(2)")
+                                .build())
+                        .testFunctionalitySingle(CertificationResultTestFunctionality.builder()
+                                .name("(b)(2)(i)(E)")
+                                .testFunctionalityId(11l)
+                                .build())
+                        .build())
+                .build();
+
+        CertifiedProductSearchDetails updatedListing = CertifiedProductSearchDetails.builder()
+                .certificationResult(CertificationResult.builder()
+                        .id(5L)
+                        .criterion(CertificationCriterion.builder()
+                                .id(17l)
+                                .certificationEdition("2015")
+                                .certificationEditionId(3L)
+                                .number("170.315 (b)(2)")
+                                .build())
+                        .testFunctionalitySingle(CertificationResultTestFunctionality.builder()
+                                .name("(b)(2)(i)(E)")
+                                .testFunctionalityId(11l)
+                                .build())
+                        .build())
+                .build();
+
+        // Run
+        reviewer.review(existingListing, updatedListing);
+
+        // Check
+        Assert.assertEquals(0, updatedListing.getErrorMessages().size());;
+    }
+
+    @Test
+    public void review_AfterRuleDateAndUserIsAdminAndAddNonRestrictedTestFunctionality_NoErrorMessages() {
+        // Setup
```

Comment:
Just because it came to mind; absolutely no reason to change anything. I've always liked the "triple A" for tests

1) Arrange
1) Act
1) Assert